qiita_pet/handlers/api_proxy/sample_template.py

```diff
@@ -0,0 +1,242 @@
+                                 QiitaDBDuplicateError, QiitaDBError,
+                                 QiitaDBDuplicateHeaderError)
+from qiita_ware.metadata_pipeline import (
+    create_templates_from_qiime_mapping_file)
+from qiita_ware.exceptions import QiitaWareError
+from qiita_pet.util import convert_text_html
+from qiita_pet.handlers.api_proxy.util import check_access
+
+
+def sample_template_info(samp_id, user_id):
+    """Equivalent to GET request to `/study/(ID)/sample_template'
+
+    Parameters
+    ----------
+    samp_id : int
+        SampleTemplate id to get info for
+    user_id : str
+        User requesting the sample template info
+
+    Returns
+    -------
+    dict of list of tuples
+        Dictionary object where the keys are the metadata categories
+        and the values are list of tuples. Each tuple is an observed value in
+        the category and the number of times its seen.
+        Format {num_samples: value,
+                category: [(val1, count1), (val2, count2), ...], ...}
+    """
+    access_error = check_access(samp_id, user_id)
+    if access_error:
+        return access_error
+    template = SampleTemplate(int(samp_id))
+    df = template.to_dataframe()
+    out = {'num_samples': df.shape[0],
+           'summary': {}}
+
+    # drop the samp_id column if it exists
+    if 'study_id' in df.columns:
+        df.drop('study_id', axis=1, inplace=True)
+    cols = list(df.columns)
+    for column in cols:
+        counts = df[column].value_counts()
+        out['summary'][str(column)] = [(str(key), counts[key])
+                                       for key in natsorted(counts.index)]
+
+    return out
+
+
+@execute_as_transaction
+def process_sample_template(study_id, user_id, data_type, sample_template):
+    """Equivalent to POST request to `/study/(ID)/sample_template'
+
+    Parameters
+    ----------
+    study_id : int
+        The current study object id
+    user_id : int
+        The current user object id
+    data_type : str
+        Data type for the sample template
+    sample_template : str
+        filepath to use for creation
+
+    Raises
+    ------
+    HTTPError
+        If the sample template file does not exists
+    """
+    access_error = check_access(int(study_id), user_id)
+    if access_error:
+        return access_error
+    # Get the uploads folder
+    _, base_fp = get_mountpoint("uploads")[0]
+    # Get the path of the sample template in the uploads folder
+    fp_rsp = join(base_fp, str(study_id), sample_template)
+
+    if not exists(fp_rsp):
+        # The file does not exist, fail nicely
+        return {'status': 'error',
+                'error': 'filepath does not exist',
+                'filepath': sample_template}
+
+    # Define here the message and message level in case of success
+    msg = ''
+    status = 'success'
+    is_mapping_file = looks_like_qiime_mapping_file(fp_rsp)
+    study = Study(int(study_id))
+    try:
```

Comment:
All the code in here to update, create, and delete was copied over from the existing study description page. I can refactor it if we want, but for now I thought we just needed the new UI working.